src/lib.rs

```diff
@@ -200,13 +239,44 @@ impl CachedQueries {
                 !attempted_parsings.is_empty(),
                 "Should've tried to parse in at least one language or else should've already failed on no candidate files"
             );
-            return Err(Error::NoSuccessfulQueryParsing(attempted_parsings));
+            if let Some((_, capture_index_error)) =
```

Comment:
I think the storage of "either a query-parsing error or a (subsequent) capture-index-resolution error" in `CachedQueries` makes sense, but here this gets a little clunky probably as far as deciding how to "resolve"/"surface" those

(eg should it be "fatal" if for one of the per-language queries there was a capture index error (eg if they passed a capture-name via `Args` and it wasn't valid for all of them? Speaking of which, probably `query_per_language` should be modeled as taking a capture name per language too, it just happens that for the `tree-sitter-lint` use case there's one capture name for all of the different language queries so it works to pass the single capture-name via `Args`)?)